src/main/java/org/apache/onami/persist/ResourceLocalTransactionFacadeFactory.java

```diff
@@ -36,6 +41,8 @@
    */
   private final EntityManagerProvider emProvider;
 
+  private final ThreadLocal<Outer> outerTransactionFacade = new ThreadLocal<>();
```

Comment:
JPA does allow multiple nested transactions, but look at where this is set -- the code detects nested transactions specially and creates an `Outer` or `Inner` as appropriate. Or do you see something more subtle?